pandas/tests/tslibs/test_conversion.py

```diff
@@ -66,3 +68,24 @@ def test_length_zero_copy(dtype, copy):
     arr = np.array([], dtype=dtype)
     result = conversion.ensure_datetime64ns(arr, copy=copy)
     assert result.base is (None if copy else arr)
+
+
+class SubDatetime(datetime):
+    pass
+
+
+@pytest.mark.parametrize("dt, expected", [
+    pytest.param(Timestamp("2000-01-01"),
+                 Timestamp("2000-01-01", tz=UTC), id="timestamp"),
+    pytest.param(datetime(2000, 1, 1),
+                 datetime(2000, 1, 1, tzinfo=UTC),
+                 id="datetime"),
+    pytest.param(SubDatetime(2000, 1, 1),
+                 SubDatetime(2000, 1, 1, tzinfo=UTC),
+                 id="subdatetime")])
```

Comment:
can this be "subclassed_datetime" or something that can't be confused with "subtraction of datetime"